### PR TITLE
Use new hfc account endpoint

### PR DESCRIPTION
**Pull request by davidkelly**
Eliminated hardcoded salt, we now get that from the server via a new call to the metaverse.  While at it, we get the iv and ckey we use to encrypt the security image (and soon maybe public keys) from the server too.  Also now really using the salt with the passphrase, which we'd not bothered with until now.

### CODE REVIEW

**Comment by davidkelly** [comment]
Oh -- of course this means your existing wallet will not decrypt, so it will tell you the wallet needs to be setup.  Note this is really not a great feature, as we wipe the old one out.  To deal with this issue -- say we don't get a response from the server, or whatever -- it is probably best to design some informational popup saying you have a wallet but we cannot decrypt it, and ask what to do next, etc...  Worth pondering and discussing...

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3141/HighFidelity-Beta-PR11325-efdc340c5c0452e9116f662888a74515852fa922.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3141/HighFidelity-Beta-PR11325-efdc340c5c0452e9116f662888a74515852fa922.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3143/HighFidelity-Beta-PR11325-bb07a586ca7181f6aed6289aa75cd02208068482.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3143/HighFidelity-Beta-PR11325-bb07a586ca7181f6aed6289aa75cd02208068482.exe

File: interface/resources/qml/hifi/commerce/wallet/Wallet.qml [hunk 1]
```diff
@@ -33,12 +33,19 @@ Rectangle {
     Hifi.QmlCommerce {
         id: commerce;
 
+        onAccountResult: {
+            if(result.status === "success") {
```
Comment (howard-stearns) [inline interface/resources/qml/hifi/commerce/wallet/Wallet.qml]: coding standard: space after if

Comment (davidkelly) [reply interface/resources/qml/hifi/commerce/wallet/Wallet.qml]: doh!  nice catch -- will update this morning

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3151/HighFidelity-Beta-PR11325-45c8dd9dd4374901a3b9d705024a96929e164c61.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3151/HighFidelity-Beta-PR11325-45c8dd9dd4374901a3b9d705024a96929e164c61.dmg

**Pull request merged**